client/jetpack-connect/signup.js

```diff
@@ -212,7 +262,128 @@ export class JetpackSignup extends Component {
 			</LoggedOutFormLinks>
 		);
 	}
+
+	renderWooDnaFooter( footerLinks ) {
+		const { authQuery } = this.props;
+		footerLinks.push(
+			<LoggedOutFormLinkItem key="back" href={ authQuery.redirectAfterAuth }>
+				<Gridicon size={ 18 } icon="arrow-left" />{ ' ' }
+				{
+					// translators: eg: Return to The WordPress.com Blog
+					this.props.translate( 'Return to %(sitename)s', {
+						args: { sitename: decodeEntities( authQuery.blogname ) },
+					} )
+				}
+			</LoggedOutFormLinkItem>
+		);
+		return <LoggedOutFormLinks>{ footerLinks }</LoggedOutFormLinks>;
+	}
+
+	renderWooDna() {
+		const { authQuery, isFullLoginFormVisible, translate, usernameOrEmail } = this.props;
+		const {
+			isCreatingAccount,
+			signUpUsernameOrEmail,
+			loginSocialConnect,
+			loginTwoFactorAuthType,
+		} = this.state;
+		let header, subHeader, content;
+		const footerLinks = [];
+		const email = signUpUsernameOrEmail || usernameOrEmail || authQuery.userEmail;
+		const wooDna = this.getWooDnaConfig();
+		let pageTitle;
+
+		if ( 'login' === this.state.wooDnaFormType ) {
+			if ( isFullLoginFormVisible ) {
+				header = translate( 'Log in to your WordPress.com account' );
+				/* translators: pluginName is the name of the Woo extension that initiated the connection flow */
+				subHeader = translate(
+					'Your account will enable you to start using the features and benefits offered by %(pluginName)s',
+					{
+						args: {
+							pluginName: wooDna.name( translate ),
+						},
+					}
+				);
+				pageTitle = translate( 'Login to WordPress.com' );
+				footerLinks.push(
+					<LoggedOutFormLinkItem key="signup" onClick={ this.showWooDnaSignupView }>
+						{ this.props.translate( 'Create a new account' ) }
+					</LoggedOutFormLinkItem>
+				);
+				footerLinks.push(
+					<LoggedOutFormLinkItem
+						key="lostpassword"
+						href={ addQueryArgs(
+							{ action: 'lostpassword' },
+							login( { locale: this.props.locale } )
+						) }
+					>
+						{ this.props.translate( 'Lost your password?' ) }
+					</LoggedOutFormLinkItem>
+				);
+			} else {
+				header = wooDna.name( translate );
+				subHeader = translate( 'Enter your email address to get started' );
+				pageTitle = translate( 'Connect' );
```

Comment:
As I understand this is our starting point when user is not logged in. But it does not allow to login with socials and if socials associated email is entered the user is redirected to passwordless flow (see gif below). IMO we should have the view similar to the [wp.com sign-in](https://wordpress.com/log-in) with input field and socials options. Note there is a message regarding new account created in case of first time login with socials. I know it's not by flow design, so cc @LevinMedia for the design feedback.

![woo_dna_google_login](https://user-images.githubusercontent.com/3139099/84166744-fec24580-aa7d-11ea-942a-c3ab08d3e294.gif)

Reply:
Extra note here at the first step (when email or username input is shown alone):

* If I add non-existing email or userId and try to use social buttons I receive error https://cloudup.com/cXPGtrSWpIv.
* If I add existing email (not matching my social email) and then use social buttons I successfully forwarded to KYC flow.

Reply:
This is becoming quite a rabbit-hole :)

Let's address the passwordless and Social login in a different branch. This PR is already massive, so as long as these changes form a cohesive experience and they don't break the existing flows, I think we should merge it sooner rather than later.

Reply:
> If I add non-existing email or userId and try to use social buttons I receive error https://cloudup.com/cXPGtrSWpIv.

I've fixed this on https://github.com/Automattic/wp-calypso/pull/43269

> if socials associated email is entered the user is redirected to passwordless flow (see gif below).

I'll work on this next. It's not a deal-breaker because, even though the redirect to wp.com sign-in is annoying, the user also correctly receives an email with the login link, so they can finish the flow.